src/test/java/com/commercetools/project/sync/service/impl/CustomObjectServiceImplTest.java

```diff
@@ -0,0 +1,206 @@
+package com.commercetools.project.sync.service.impl;
+
+import static java.util.Collections.singletonList;
+import static java.util.Optional.empty;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+import com.commercetools.project.sync.model.LastSyncCustomObject;
+import com.commercetools.project.sync.service.CustomObjectService;
+import com.commercetools.sync.products.helpers.ProductSyncStatistics;
+import io.sphere.sdk.client.BadGatewayException;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.customobjects.CustomObject;
+import io.sphere.sdk.customobjects.commands.CustomObjectUpsertCommand;
+import io.sphere.sdk.customobjects.queries.CustomObjectQuery;
+import io.sphere.sdk.queries.PagedQueryResult;
+import io.sphere.sdk.utils.CompletableFutureUtils;
+import java.time.ZonedDateTime;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import org.junit.jupiter.api.Test;
+
+class CustomObjectServiceImplTest {
+
+  @Test
+  @SuppressWarnings("unchecked")
+  void getCurrentCtpTimestamp_OnSuccessfulUpsert_ShouldCompleteWithCtpTimestampMinusTwoMinutes() {
+    // preparation
+    final SphereClient client = mock(SphereClient.class);
```

Comment:
these parts are common for tests, maybe it's better if you moved to a @BeforeAll hook or not ?
```
 final SphereClient client = mock(SphereClient.class);
 final CustomObject<String> stringCustomObject = mock(CustomObject.class);
 when(stringCustomObject.getLastModifiedAt()).thenReturn(ZonedDateTime.now());
```

Reply:
38cfdf6